### PR TITLE
Make the contextMenu in StructurePanel und in GalleryPanel identical

### LINKED ISSUE #3646: "Assign to next element" not available in the Erweiterter Strukturbaum
### Problem
In the Erweiterter Strukturbaum, it is not possible, to assign an image to two structure elements with the function "Assign to next element". 
If the links were created, in the Einfacher Strukturbaum, they are shown in the gallery. 

### Solution 
It should be possible to add the links in the physical structure by context menu. additional to 
- Add element
- Remove element

it should be offered: 
- Assign to next element

Otherwise, the structure tree cannot be used. 

### Example

assignelement01

### PR DESCRIPTION
**Pull request by IkramMaalej**
Fixes #3646

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/AddDocStrucTypeDialog.java [hunk 1]
```diff
@@ -418,6 +419,19 @@ public void prepare() {
...
     }
 
+    private void checkSelectedLogicalNode() {
+        //If a view is selected in logical tree then the 'selectedLogicalNode' will be set to the parent of this view
+        TreeNode selectedLogicalNode = dataEditor.getStructurePanel().getSelectedLogicalNode();
+        if (Objects.nonNull(selectedLogicalNode) && selectedLogicalNode.getData() instanceof StructureTreeNode) {
+            StructureTreeNode structureTreeNode = (StructureTreeNode) selectedLogicalNode.getData();
+            if (structureTreeNode.getDataObject() instanceof View) {
+                if (Objects.nonNull(selectedLogicalNode.getParent())) {
+                    dataEditor.getStructurePanel().setSelectedLogicalNode(selectedLogicalNode.getParent());
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/AddDocStrucTypeDialog.java:429]: I am not sure about this line: I understand that the selected logical node must be set to the parent structure when a page is selected, but what happens when the user then clicks `Cancel`? Then the `selectedLogicalNode` in the `StructurePanel` is still set to the parent structure, even though that is not visible in the tree, where the page still seems to be selected.

File: Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataEditor/gallery.xhtml [hunk 2]
```diff
@@ -35,6 +35,7 @@
                                      imagePreviewForm:mediaViewData
                                      logicalTree
                                      physicalTree
+                                     mediaContextMenu
```
Comment (solth) [inline Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataEditor/gallery.xhtml:38]:
```suggestion
                                     imagePreviewForm:mediaContextMenu
```
Shouldn't this have the `imagePreviewForm:` prefix like the other `update`s?

File: Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataEditor/logicalStructure.xhtml [hunk 3]
```diff
@@ -72,15 +75,17 @@
...
                         icon="fa fa-plus fa-sm"
                         styleClass="plain"
                         disabled="#{readOnly}"
                         action="#{DataEditorForm.addDocStrucTypeDialog.prepare}"
                         actionListener="#{DataEditorForm.addDocStrucTypeDialog.resetValues}"
                         process="@this"
                         oncomplete="PF('dialogAddDocStrucType').show()"
-                        update="dialogAddDocStrucTypeDialog"/>
+                        update="dialogAddDocStrucTypeDialog">
+            <f:actionListener binding="#{DataEditorForm.addDocStrucTypeDialog.preparePreselectedViews()}"/>
```
Comment (solth) [inline Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataEditor/logicalStructure.xhtml:87]:
```suggestion
                     <f:actionListener binding="#{DataEditorForm.addDocStrucTypeDialog.preparePreselectedViews()}"/>
```

**Review by solth — COMMENTED** [review]
This is a rather small technical change, but I believe it will help working with the metadata editor considerably! 👍 

Just one small nitpick: the context menu in the gallery does not disappear when clicking somewhere else, for example when selecting a different stripe:
Bildschirmfoto 2021-03-31 um 11 29 15

**Comment by Kathrin-Huber** [comment]
please resolve conflicts

File: Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataEditor/gallery.xhtml [hunk 4]
```diff
@@ -357,8 +359,34 @@
...
                             update="dialogAddDocStrucTypeDialog">
                     <f:actionListener binding="#{DataEditorForm.addDocStrucTypeDialog.preparePreselectedViews()}"/>
-                    <f:actionListener binding="#{DataEditorForm.structurePanel.setSelectedLogicalNode(DataEditorForm.structurePanel.logicalTree.children.get(DataEditorForm.structurePanel.logicalTree.childCount - 1))}"/>
                 </p:menuitem>
+                <p:menuitem value="#{msgs.assignToNextElement}"
+                            icon="fa fa-link fa-sm"
+                            styleClass="plain"
+                            disabled="#{readOnly}"
+                            rendered="#{DataEditorForm.structurePanel.assignableSeveralTimes and not
+                            DataEditorForm.structurePanel.assignedSeveralTimes and DataEditorForm.selectedMedia.size() eq 1}"
```
Comment (solth) [inline Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataEditor/gallery.xhtml:368]:
```suggestion
                            rendered="#{DataEditorForm.structurePanel.assignableSeveralTimes and
                            DataEditorForm.selectedMedia.size() eq 1}"
```

**Review by solth — CHANGES_REQUESTED** [review]
Just one little change to completely synchronize context menus in structure tree and gallery since #4319 has been merged in the meantime.

**Review by solth — APPROVED** [review]

**Pull request merged**